code/modules/admin/verbs/adminsay.dm

```diff
@@ -45,6 +45,48 @@
 		var/msg = input(src, null, "msay \"text\"") as text|null
 		cmd_mentor_say(msg)
 
+/client/proc/get_dev_team_say()
+	if(check_rights(R_DEV_TEAM | R_ADMIN | R_MOD))
+		var/msg = input(src, null, "devsay \"text\"") as text|null
+		cmd_dev_say(msg)
+
+/client/proc/cmd_dev_say(msg as text)
+	set name = "Devsay"
+	set hidden = TRUE
+
+	if(!check_rights(R_ADMIN|R_DEV_TEAM)) // Catch any non-admins trying to use this proc
+		return
+
+	msg = emoji_parse(copytext_char(sanitize(msg), 1, MAX_MESSAGE_LEN))
+	log_devsay(msg, src)
+	var/datum/say/devsay = new(usr.ckey, usr.client.holder.rank, msg, world.timeofday)
+	GLOB.devsays += devsay
+	mob.create_log(OOC_LOG, "DEVSAY: [msg]")
+
+	if(!msg)
+		return
+
+	// Do this up here before it gets sent to everyone & emoji'd
+	if(SSredis.connected)
+		var/list/data = list()
+		data["author"] = usr.ckey
+		data["source"] = GLOB.configuration.system.instance_id
+		data["message"] = html_decode(msg)
+		SSredis.publish("byond.devsay", json_encode(data))
```

Comment:
Todo - Alice update.

Dunno if we need #devsay, probably to not bridge it all to the main chat